multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/MtaArchiveContentResolver.java

```diff
@@ -0,0 +1,137 @@
+import org.cloudfoundry.multiapps.controller.process.Messages;
+import org.cloudfoundry.multiapps.mta.model.DeploymentDescriptor;
+import org.cloudfoundry.multiapps.mta.model.Module;
+import org.cloudfoundry.multiapps.mta.model.RequiredDependency;
+import org.cloudfoundry.multiapps.mta.model.Resource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class MtaArchiveContentResolver {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MtaArchiveContentResolver.class);
+
+    private final DeploymentDescriptor descriptor;
+
+    private final ApplicationConfiguration configuration;
+
+    private final ContentLengthTracker sizeTracker;
+
+    private final ExternalFileProcessor fileProcessor;
+
+    public MtaArchiveContentResolver(DeploymentDescriptor descriptor, ApplicationConfiguration configuration, ExternalFileProcessor fileProcessor, ContentLengthTracker sizeTracker) {
+        this.descriptor = descriptor;
+        this.configuration = configuration;
+        this.fileProcessor = fileProcessor;
+        this.sizeTracker = sizeTracker;
+    }
+
+    public void resolveMtaArchiveFilesInDescriptor(String space, String appArchiveId, MtaArchiveHelper helper) {
+        resolveResourcesContent(space, appArchiveId, helper);
+        resolveRequiresDependenciesContent(space, appArchiveId, helper);
+        long totalSizeOfResolvedEntries = sizeTracker.getTotalSize();
+        LOGGER.debug(MessageFormat.format(Messages.TOTAL_SIZE_OF_ALL_RESOLVED_CONTENT_0, totalSizeOfResolvedEntries));
+        if (totalSizeOfResolvedEntries > configuration.getMaxResolvedExternalContentSize()) {
+            throw new SLException(Messages.ERROR_RESOLVED_FILE_CONTENT_IS_0_WHICH_IS_LARGER_THAN_MAX_1, totalSizeOfResolvedEntries,
+                                  configuration.getMaxResolvedExternalContentSize());
+        }
+    }
+
+    private void resolveResourcesContent(String space, String appArchiveId, MtaArchiveHelper helper) {
+        Map<String, List<String>> resourceFileAttributes = helper.getResourceFileAttributes();
+        for (var entry : resourceFileAttributes.entrySet()) {
+            Map<String, Object> fileContentForEntry = fileProcessor.processFileContent(space, appArchiveId, entry.getKey());
+            mergeResourcesFromFile(entry, fileContentForEntry);
+        }
+    }
+
+    private void mergeResourcesFromFile(Map.Entry<String, List<String>> entry, Map<String, Object> fileContent) {
+        List<Resource> list = new ArrayList<>();
+        for (Resource resource : descriptor.getResources()) {
+            if (entryMatchesResource(entry, resource)) {
+                setResourceParametersFromFile(resource, fileContent);
+                sizeTracker.incrementFileSize();
```

Comment:
I think the check for the size should be here so it can be guaranteed it will throw exception as early as possible

Reply:
Done